multimodal/src/autogluon/multimodal/utils/object_detection.py

```diff
@@ -658,12 +725,379 @@ def get_coco_format_classes(sample_data_path):
+    idx2classname = {i: classname for (i, classname) in enumerate(detection_classes)}
+
+    for image_pred, image_name in zip(pred, image_names):
+        box_info = []
+        for class_idx, bboxes in enumerate(image_pred):
+            pred_class = idx2classname[class_idx]
+
+            for bbox in bboxes:
+                box_info.append({"class": pred_class, "bbox": list(bbox[:4]), "score": bbox[4]})
+        results.append([image_name, box_info])
+    result_df = pd.DataFrame(results, columns=["image", "bboxes"])
+    result_df.to_csv(result_path, index=False)
+    logger.info("Saved detection results to {}".format(result_path))
+    print("Saved detection results to {}".format(result_path))
+
+
+def save_result_df_deprecated(pred: Iterable, data: pd.DataFrame, result_path: str, detection_data_path: str = None):
+    """
+    Saving detection results in pd.DataFrame format (per detection)
+
+    Parameters
+    ----------
+    pred
+        List containing detection results for one image
+    data
+        pandas data frame containing the image information to be tested
+    result_path
+        path to save result
+    detection_data_path
+        The data to make predictions for. Should contain same column names as training data and
+        follow same format (except for the `label` column).
+        None if inference on custom data (user provided)
+    Returns
+    -------
+    None
+    """
+    image_names = data["image"].to_list()
+    results = []
+    if detection_data_path:
+        detection_classes = get_detection_classes(detection_data_path)
+        idx2classname = {i: classname for (i, classname) in enumerate(detection_classes)}
+    else:
+        idx2classname = None
+    for image_pred, image_name in zip(pred, image_names):
+        for class_idx, bboxes in enumerate(image_pred):
+            if idx2classname:
+                pred_class = idx2classname[class_idx]
+            else:
+                pred_class = class_idx
+            for bbox in bboxes:
+                row = [image_name, pred_class, list(bbox[:4]), bbox[4]]
+                results.append(row)
+    result_df = pd.DataFrame(results, columns=["image", "class", "bbox", "score"])
+    result_df.to_csv(result_path, index=False)
+    logger.info(25, "Saved detection results to {}".format(result_path))
+    # Mark: Do we want to print to the user and let them know where the results are saved?
+    print("Saved detection results to {}".format(result_path))
+
+
+def save_result_coco_format(detection_data_path, pred, result_path):
+    coco_dataset = COCODataset(detection_data_path)
+    result_name, _ = os.path.splitext(result_path)
+    result_path = result_name + ".json"
+    coco_dataset.save_result(pred, from_coco_or_voc(detection_data_path, "test"), save_path=result_path)
+    logger.info(25, f"Saved detection result to {result_path}")
+    print(f"Saved detection result to {result_path}")
+
+
+def save_voc_result_to_coco_format(pred, data, result_path):
```

Comment:
If we don't have this function, either remove it or add a TODO here saying why we need this function.

Reply:
This is deleted